File: Python/Fundamentals/04_Functions/Exercise/password_validator.py
input_password = input()


def password_validator(password):
    validator1 = False
    validator2 = False
    validator3 = False
    if not 6 <= len(password) <= 10:
        print('Password must be between 6 and 10 characters')
    else:
        validator1 = True

    if not password.isalnum():
        print('Password must consist only of letters and digits')
    else:
        validator2 = True

    digit_count = 0
    for symbol in range(0, len(password)):
        if password[symbol].isdigit():
            digit_count += 1

    if not digit_count >= 2:
        print('Password must have at least 2 digits')
    else:
        validator3 = True

    if validator1 and validator2 and validator3:
        print('Password is valid')


password_validator(input_password)

File: JS/JS  Fundamentals/04_Functions/exercise/passwordValidator.js
function solve(password) {
    const result = [];
    let countOfDigit = 0;

    if (password.length < 6 || password.length > 10){
        result.push('Password must be between 6 and 10 characters');
    }

    for (let char of password) {
        if (char.toLowerCase() == char.toUpperCase() && isNaN(Number(char))) {
            result.push('Password must consist only of letters and digits');
            break;
        }
    }

    for (let char of password) {
        if (!isNaN(Number(char))){
            countOfDigit += 1;
        }
    }

    if (countOfDigit < 2){
        result.push('Password must have at least 2 digits');
    }

    if (result.length == 0) {
        console.log('Password is valid');
    } else {
        console.log(result.join('\n'));
    }
}

solve('logIn');
console.log('--------------');
solve('MyPass123');
console.log('--------------');
solve('Pa$s$s')

